### PR TITLE
Make Path.Build.chmod more flexible

### PR DESCRIPTION
**Pull request by snowleopard**
I'm factoring out a few smaller patches out of my upcoming PR on rewriting the shared cache.

This patch removes some code duplication and makes `Path.Build.chmod` more flexible.

### CODE REVIEW

File: otherlibs/stdune-unstable/path.ml [hunk 1]
```diff
@@ -610,17 +610,21 @@ module Kind = struct
     | External x -> External (External.relative x (Local.to_string y))
 end
 
-let chmod_generic ~mode ?(op = `Set) path =
+let chmod_generic ~mode ?(stats = None) ?(op = `Set) path =
```
Comment (jeremiedimino) [inline otherlibs/stdune-unstable/path.ml:613]: Not specific to this PR, but `?(stats = None)` is the same as  `?stats`

**Review by jeremiedimino — APPROVED** [review]
Looks good. But thinking about it, I'm wondering if we shouldn't only have a function to set permissions rather than support all 3 operations. Just to avoid calling `Unix.stat` twice by accident.

**Comment by snowleopard** [comment]
> Looks good. But thinking about it, I'm wondering if we shouldn't only have a function to set permissions rather than support all 3 operations. Just to avoid calling `Unix.stat` twice by accident.

I agree, that seems better. Let me try re-doing it this way.

File: otherlibs/stdune-unstable/path.ml (hunk 1 above)
Comment (jeremiedimino) [reply otherlibs/stdune-unstable/path.ml:613]: Oh! Ok then this is this type that's too much: `?stats:Unix.stats option`. Just `?stats:Unix.stats` would do.

Comment (jeremiedimino) [reply otherlibs/stdune-unstable/path.ml:613]: But, well doesn't matter if we change this to only support setting

**Comment by snowleopard** [comment]
@jeremiedimino I made the type more precise:

```
  val chmod :
       mode:int
    -> ?op:[ `Set | `Add of Unix.stats option | `Remove of Unix.stats option ]
    -> t
    -> unit
```

Now there is just no way to provide unneeded stats in the `Set` case.

**Comment by snowleopard** [comment]
After talking to @jeremiedimino, I simplified this even further.

**Review by jeremiedimino — APPROVED** [review]
Looks much simpler now!

**Pull request merged**